addons/point_of_sale/migrations/13.0.1.0.1/end-migration.py

```diff
@@ -5,7 +5,57 @@
 
 
 def create_pos_payment_methods(env):
-    env["pos.config"].post_install_pos_localisation()
+    """We will create a Payment method for each journal used on configurations"""
+    if not openupgrade.column_exists(
+            env.cr, "pos_payment_method", "old_journal_id"):
+        openupgrade.logged_query(
+            env.cr, """
+            ALTER TABLE pos_payment_method
+            ADD COLUMN old_journal_id int"""
+        )
+        env.cr.execute("""
+               SELECT DISTINCT journal_id
+               FROM pos_config_journal_rel
+           """)
+        journal_ids = [aj[0] for aj in env.cr.fetchall()]
+        if journal_ids:
+            openupgrade.logged_query(env.cr, """
+                    INSERT INTO pos_payment_method
+                        (
+                            old_journal_id, 
+                            name, 
+                            receivable_account_id,
+                            is_cash_count,
+                            cash_journal_id,
+                            company_id,
+                            create_uid,
+                            create_date,
+                            write_uid,
+                            write_date
+                        )
+                    SELECT 
+                        aj.id, 
+                        aj.name,
+                        rc.account_default_pos_receivable_account_id,
+                        aj.type = 'cash',
+                        CASE WHEN aj.type = 'cash' THEN aj.id ELSE NULL END,
+                        rc.id,
+                        aj.create_uid,
+                        aj.create_date,
+                        aj.write_uid,
+                        aj.write_date
+                    FROM account_journal aj
+                    INNER JOIN res_company rc ON rc.id = aj.company_id
+                    WHERE aj.id in %s
+                """, (tuple(journal_ids),))
+        openupgrade.logged_query(
+            env.cr, """
+                INSERT INTO pos_config_pos_payment_method_rel
+                    (pos_config_id, pos_payment_method_id)
+                SELECT pos_config_id, ppm.id
+                FROM pos_config_journal_rel pcjr
+                JOIN pos_payment_method ppm ON pcjr.journal_id = ppm.old_journal_id
+                """)
 
```

Comment:
```suggestion
   env["pos.config"].generate_pos_journal()
```
?

Reply:
That is a bad approach IMO. I know it is easier, but it will not reflect reality. In my case, we have 4 Payment methods, with this change only one will be kept

Reply:
Ups, sorry, I correct myself. I though you said assign_payment_journals.

I might be wrong, let me check it again.

Reply:
Yes you were right, sorry